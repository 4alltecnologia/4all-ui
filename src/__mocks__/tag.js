export const tags = {
  list: [
    {
      id: 1,
      title: 'Tag1',
    },
    {
      id: 2,
      title: 'Tag2',
    },
    {
      id: 3,
      title: 'Tag3',
    },
    {
      id: 4,
      title: 'Tag4',
    },
    {
      id: 5,
      title: 'Tag5',
    },
  ],
  isLoading: false,
};

export const emptyTags = {
  list: [],
  isLoading: false,
};

export const selectedTags = [1];

export const singleTag = {
  id: 1,
  label: 'Tag1',
};
