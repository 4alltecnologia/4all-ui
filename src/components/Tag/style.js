import styled, { css } from 'styled-components';
import { fontStyle, flexPosition, globalStyle } from '../../styles/mixins';
import theme from '../../styles/variables';
import { Icon } from '../Icons/Close';

const errorTag = css`
  color: ${theme.colors.GRAY_LIGHT};
  background-color: ${theme.colors.DANGER_COLOR};
  border-color: ${theme.colors.DANGER_COLOR};
`;

const secondaryTag = css`
  color: ${theme.colors.GRAY_DARKER};
  background-color: transparent;
  border-color: ${theme.colors.GRAY_LIGHT_2};

  &:hover {
    background-color: transparent;
  }
`;

const disabledStyle = css`
  background-color: ${theme.colors.GRAY_LIGHT_2};
  border-color: ${theme.colors.GRAY_LIGHT_2};
  color: ${theme.colors.GRAY_MEDIUM_3};

  &:hover {
    background-color: ${theme.colors.GRAY_LIGHT_2};
  }
`;

export const CustomTag = styled.div`
  ${globalStyle()}
  background-color: ${({ bgColor }) => bgColor};
  border: ${({ border }) => border};
  border-radius: 20px;
  color: ${({ color }) => color || theme.colors.WHITE};
  ${flexPosition({ align: 'center', justify: 'center' })}
  ${fontStyle(
    ({ color }) => color,
    theme.fontSizes.SIZE_7,
    theme.fontWeights.SEMIBOLD,
  )}
  height: ${({ height }) => height || 'auto'};
  line-height: 17px;
  margin-right: 10px;
  margin-bottom: 10px;
  min-width: 50px;
  padding: ${({ removeTag }) => (removeTag ? '0 5px 0 15px' : '3px 15px')};
  position: relative;
  width: ${({ width }) => width || 'auto'};

  &:hover {
    ${({ hoverStyle }) => hoverStyle};
  }

  &:last-child {
    margin-right: 0;
  }

  ${Icon} {
    flex-shrink: 0;
    margin: -1px 0 -1px 10px;
  }
  
  ${({ removeTag }) => (removeTag ? 'justify-content: space-between' : {})}
  ${({ error }) => (error ? errorTag : {})};
  ${({ secondary }) => (secondary ? secondaryTag : {})};
  ${({ disabled }) => (disabled ? disabledStyle : {})};
  ${({ customStyles }) => customStyles};
`;
