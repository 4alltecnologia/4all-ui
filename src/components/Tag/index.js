import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import * as styles from './style';
import theme from '../../styles/variables';
import CloseIcon from '../Icons/Close';
import { GlobalStyle } from '../../styles/mixins';

const getIconColor = ({ iconColor, disabled, secondary }) => {
  if (iconColor) {
    return iconColor;
  }

  if (disabled || secondary) {
    return theme.colors.GRAY_MEDIUM_3;
  }

  return theme.colors.white;
};

const Tag = ({
  id,
  tag,
  className,
  color,
  height,
  width,
  bgColor,
  border,
  disabled,
  customStyles,
  onHoverTag,
  onBlurTag,
  onClick,
  onRemoveTag,
  children,
  secondary,
  error,
  iconColor,
}) => (
  <Fragment>
    <GlobalStyle />
    <styles.CustomTag
      id={id}
      className={`tag-4all ${className}`}
      color={color}
      height={height}
      width={width}
      bgColor={bgColor}
      disabled={disabled}
      customStyles={customStyles}
      border={border}
      onMouseLeave={onBlurTag ? () => onBlurTag(tag) : null}
      onMouseOver={onHoverTag ? () => onHoverTag(tag) : null}
      onFocus={onHoverTag ? () => onHoverTag(tag) : null}
      onClick={onClick}
      removeTag={!!onRemoveTag}
      secondary={secondary}
      error={error}
    >
      {children}
      {!!onRemoveTag && (
        <CloseIcon
          color={getIconColor({
            iconColor,
            disabled,
            secondary,
          })}
          onClick={onRemoveTag}
        />
      )}
    </styles.CustomTag>
  </Fragment>
);

Tag.propTypes = {
  id: PropTypes.string,
  tag: PropTypes.object,
  className: PropTypes.string,
  color: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  bgColor: PropTypes.string,
  border: PropTypes.string,
  onHoverTag: PropTypes.func,
  onBlurTag: PropTypes.func,
  disabled: PropTypes.bool,
  customStyles: PropTypes.object,
  onClick: PropTypes.func,
  onRemoveTag: PropTypes.func,
  children: PropTypes.any.isRequired,
  secondary: PropTypes.bool,
  error: PropTypes.bool,
  iconColor: PropTypes.string,
};

Tag.defaultProps = {
  id: null,
  tag: {},
  className: '',
  color: theme.colors.white,
  height: 'auto',
  width: 'auto',
  bgColor: theme.colors.MAIN_COLOR,
  disabled: false,
  border: `1px solid ${theme.colors.MAIN_COLOR}`,
  onHoverTag: null,
  onBlurTag: null,
  onClick: null,
  onRemoveTag: null,
  customStyles: {},
  secondary: false,
  error: false,
  iconColor: theme.colors.white,
};

export default Tag;
