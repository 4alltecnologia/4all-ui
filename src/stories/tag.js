import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { action } from '@storybook/addon-actions'; // eslint-disable-line import/no-extraneous-dependencies
import Tag from '../components/Tag';
import theme from '../styles/variables';

storiesOf('Tag', module)
  .add('With Close Button', () => (
    <div style={{ display: 'flex' }}>
      <Tag onRemoveTag={action('With onRemoveTag Function ')}>Label</Tag>
    </div>
  ))
  .add('Primary', () => (
    <div style={{ display: 'flex' }}>
      <Tag>Label</Tag>
    </div>
  ))
  .add('Error', () => (
    <div style={{ display: 'flex' }}>
      <Tag error>Label</Tag>
    </div>
  ))
  .add('Secondary', () => (
    <div style={{ display: 'flex' }}>
      <Tag secondary>Label</Tag>
    </div>
  ))
  .add('Disabled', () => (
    <div style={{ display: 'flex' }}>
      <Tag disabled>Label</Tag>
    </div>
  ))
  .add('Custom Style', () => (
    <div style={{ display: 'flex' }}>
      <Tag
        bgColor={theme.colors.WHITE}
        color={theme.colors.INFO_COLOR}
        border={`1px solid ${theme.colors.GRAY_MEDIUM_2}`}
      >
        Label
      </Tag>
      <Tag
        bgColor={theme.colors.WHITE}
        color={theme.colors.MAIN_COLOR}
        border={`1px solid ${theme.colors.GRAY_MEDIUM_2}`}
      >
        Label
      </Tag>
      <Tag
        bgColor={theme.colors.WHITE}
        color="black"
        border={`1px solid ${theme.colors.GRAY_MEDIUM_2}`}
      >
        Label
      </Tag>
      <Tag
        bgColor={theme.colors.WHITE}
        color={theme.colors.HIGHLIGHT_COLOR}
        border={`1px solid ${theme.colors.GRAY_MEDIUM_2}`}
      >
        Label
      </Tag>
      <Tag
        bgColor={theme.colors.WHITE}
        color={theme.colors.DANGER_COLOR}
        border={`1px solid ${theme.colors.GRAY_MEDIUM_2}`}
      >
        Label
      </Tag>
    </div>
  ));
