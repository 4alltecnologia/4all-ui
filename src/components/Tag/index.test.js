import React from 'react';
import { shallow } from 'enzyme';
import Tag from '.';
import { singleTag } from '../../__mocks__/tag';

const clickFn = jest.fn();

describe('<Tag />', () => {
  it('should render the component', () => {
    const component = shallow(
      <Tag className="tag-item" tag={singleTag} onClick={clickFn}>
        {singleTag.label}
      </Tag>,
    );

    expect(component).toMatchSnapshot();
  });

  it('should call the click function and contain string passed', () => {
    const component = shallow(
      <Tag className="tag-item" tag={singleTag} onClick={clickFn}>
        {singleTag.label}
      </Tag>,
    );
    component.find('.tag-item').simulate('click');

    expect(clickFn).toHaveBeenCalled();
    expect(component.text()).toContain(singleTag.label);
  });
});
