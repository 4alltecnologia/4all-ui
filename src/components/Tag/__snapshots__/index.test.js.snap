// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    bgColor="#4FA444"
    border="none"
    className="tag-item"
    color="#FFF"
    customStyles={null}
    disabled={false}
    error={false}
    height="auto"
    id={null}
    onBlurTag={null}
    onClick={[Function]}
    onHoverTag={null}
    onRemoveTag={null}
    secondary={false}
    tag={
        Object {
            "id": 1,
            "label": "Tag1",
          }
    }
    width="auto"
>
    Tag1
</Tag>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <GlobalStyleComponent />,
        <Unknown
          bgColor="#4FA444"
          border="none"
          className="tag-4all tag-item"
          color="#FFF"
          customStyles={null}
          disabled={false}
          error={false}
          height="auto"
          id={null}
          onClick={[Function]}
          onFocus={null}
          onMouseLeave={null}
          onMouseOver={null}
          onRemoveTag={false}
          secondary={false}
          width="auto"
>
          Tag1
</Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bgColor": "#4FA444",
          "border": "none",
          "children": Array [
            "Tag1",
            false,
          ],
          "className": "tag-4all tag-item",
          "color": "#FFF",
          "customStyles": null,
          "disabled": false,
          "error": false,
          "height": "auto",
          "id": null,
          "onClick": [Function],
          "onFocus": null,
          "onMouseLeave": null,
          "onMouseOver": null,
          "onRemoveTag": false,
          "secondary": false,
          "width": "auto",
        },
        "ref": null,
        "rendered": Array [
          "Tag1",
          false,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
  position: relative;
  ",
              "
  display: flex;
  align-items: center;
  justify-content: center;
  
",
              "
  ",
              "
  color: props => props.color;
  font-size: 14px;
  font-weight: 700;
",
              "
  min-width: 50px;
  border-radius: 20px;
  padding: ",
              [Function],
              ";
  margin-right: 10px;
  margin-bottom: 10px;
  background-color: ",
              [Function],
              ";
  color: ",
              [Function],
              ";
  height: ",
              [Function],
              ";
  width: ",
              [Function],
              ";
  border: ",
              [Function],
              ";
  &:hover {
    ",
              [Function],
              ";
  }
  ",
              [Function],
              "
  ",
              [Function],
              ";
  ",
              [Function],
              ";
  ",
              [Function],
              ";
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GlobalStyleComponent />,
          <Unknown
            bgColor="#4FA444"
            border="none"
            className="tag-4all tag-item"
            color="#FFF"
            customStyles={null}
            disabled={false}
            error={false}
            height="auto"
            id={null}
            onClick={[Function]}
            onFocus={null}
            onMouseLeave={null}
            onMouseOver={null}
            onRemoveTag={false}
            secondary={false}
            width="auto"
>
            Tag1
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bgColor": "#4FA444",
            "border": "none",
            "children": Array [
              "Tag1",
              false,
            ],
            "className": "tag-4all tag-item",
            "color": "#FFF",
            "customStyles": null,
            "disabled": false,
            "error": false,
            "height": "auto",
            "id": null,
            "onClick": [Function],
            "onFocus": null,
            "onMouseLeave": null,
            "onMouseOver": null,
            "onRemoveTag": false,
            "secondary": false,
            "width": "auto",
          },
          "ref": null,
          "rendered": Array [
            "Tag1",
            false,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
  position: relative;
  ",
                "
  display: flex;
  align-items: center;
  justify-content: center;
  
",
                "
  ",
                "
  color: props => props.color;
  font-size: 14px;
  font-weight: 700;
",
                "
  min-width: 50px;
  border-radius: 20px;
  padding: ",
                [Function],
                ";
  margin-right: 10px;
  margin-bottom: 10px;
  background-color: ",
                [Function],
                ";
  color: ",
                [Function],
                ";
  height: ",
                [Function],
                ";
  width: ",
                [Function],
                ";
  border: ",
                [Function],
                ";
  &:hover {
    ",
                [Function],
                ";
  }
  ",
                [Function],
                "
  ",
                [Function],
                ";
  ",
                [Function],
                ";
  ",
                [Function],
                ";
  ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
